examples/generic-azure.yaml

```diff
@@ -673,3 +673,39 @@ groups:
           ## References
           - section 9.1.2 https://documentation.suse.com/sbp/all/single-html/SLES4SAP-hana-sr-guide-PerfOpt-15/
         scored: true
+      - id: 1.5.3
+        description: "<sid>adm user and its <sid> group exist and enabled"
+        audit: |
+          # Get SIDs from cibadmin
+          IFS=' ' read -r -a sid_list_raw <<< $(sudo cibadmin -Q | xmllint -xpath '//primitive//nvpair[@name="SID"]/@value' -)
+          # Parses sids and makes them unique. There is possibly a more beautiful way to do it with regexp-s rather than with loops.
+          sid_list_not_unique=$(for i in "${sid_list_raw[@]}"; do echo $i | sed -e "s/.*value=\"\(...\)\".*/\1/" | tr '[:upper:]' '[:lower:]' ; done)
+          sid_list_sorted_and_unique=$(echo ${sid_list_not_unique[@]} | tr ' ' '\n' | sort -u | tr '\n' ' ')
+
+          # Check the /etc/nsswitch.conf
+          if ! cat /etc/nsswitch.conf | grep "passwd:" | grep -q "compat\|files"; then echo failed=true; fi
+          if ! cat /etc/nsswitch.conf | grep "files:" | grep -q "compat\|files"; then echo failed=true; fi
+
+          # Check the /etc/passwd and /etc/group
+          for sid in ${sid_list_sorted_and_unique}
+          do
+            if ! cat /etc/passwd | cut -d: -f1 | grep -q ${sid}adm; then echo failed=true; fi
+            if cat /etc/passwd | grep -e "^${sid}adm" | cut -d: -f7 | grep -q "nologin"; then echo failed=true; fi
+            if ! cat /etc/group | cut -d: -f1,4 | grep -q "sapsys:${sid}adm"; then echo failed=true; fi
```

Comment:
@aleksei-burlakov 
sid is 'prd'
entry in /etc/group looks like 'sapsys:x:79:briel,prdadm,ha1adm'
if ! cat /etc/group | cut -d: -f1,4 | grep -q "sapsys:${sid}adm"; then echo failed=true; fi
returns 'failed=true'
It's not so unusual that more than one user is added to a group. And if we have more than one SID defined, we may have more sidadm user added to the group sapsys.

Reply:
Thank you, good catch!

Reply:
Just as a remark. AWK can combine things like cutting into fields and comparing multiple fields very well:

awk -F: 'BEGIN { found=1 } (($1 == "sapsys") && ($2 ~ user)) { found=0 } END { exit found }' user=hugadm

Just 2 ct "hugadm" is just the <sid>adm example and awk can also open the file, so we can use 1 command instead
of 3 to 4 using pipes.